indigo/docs/guides/howto-hello-indigo-tyrian.md

```diff
@@ -0,0 +1,418 @@
+```
+
+We'll also need to tell Indigo to use the new sub-system, which we can do by
+adding an entry to the `boot` method like so:
+
+```diff
+   def boot(flags: Map[String, String]): Outcome[BootResult[Unit]] =
+     Outcome(
+       BootResult
+         .noData(config)
+         .withAssets(assets)
++        .withSubSystems(tyrianSubSystem)
+     )
+```
+
+### Create a Tyrian App
+
+We'll now develop our initial Tyrian app, which will consist simply of
+a canvas (for Indigo), a counter, and a button, wrapped in a few `div` elements.
+Create a new file called `HelloTyrian.scala` inside the `helloindigo/src` folder
+and add the following contents found
+[here](https://gist.github.com/hobnob/436318b3ae5eed5891ba2b18bb8c264b).
+
+You can now build the project in mill or sbt and then run `yarn start` to see the HelloIndigo demo running inside a Tyrian website. This is great if you want a
+website surrounding your game with no interaction, but it would be much more
+useful to get them both talking to each other... this is what we're doing next!
+
+## Tyrian to Indigo Communication
+
+The first thing we'll do is get Tyrian to request that Indigo add a new dot. To
+do this we'll need to add an `OnClick` event to our button, which will tell
+Indigo what to do.
+
+In `HelloTyrian.scala`, we'll need a new message type. Add a new message type called AddDot like so:
+
+```diff
+   enum Msg:
+     case StartIndigo extends Msg
++    case AddDot      extends Msg
+```.
+
+Next we'll need to deal with that message in Tyrian, by adding a case to `update`
+like this:
+
+```diff
+     msg match
++      case Msg.AddDot =>
++        (model, model.bridge.publish(IndigoGameId(gameDivId), 0))
+       case Msg.StartIndigo =>
+```
+
+The final part on the Tyrian side is to hook the button up to fire the `AddDot`
+message. To do this add an `OnClick` attribute to our button like so:
+
+```diff
+       div(`class` := "btn")(
+-        button()("Click me")
++        button(onClick(Msg.AddDot))("Click me")
+       )
+```
+
+This will now send an event to Indigo with an integer message, which in this case
+we've set to zero. Now we just need to modify Indigo to receive and
+process that message. To do this change the `MouseEvent.Click` line in
+`HelloIndigo.scala` with the following:
+
+```diff
+   ): GlobalEvent => Outcome[Model] = {
+-    case MouseEvent.Click(pt) =>
++    case tyrianSubSystem.TyrianEvent.Receive(msg) =>
++      val pt               = Point(100, 100)
+       val adjustedPosition = pt - model.center
+```
+
+We've also added a line below that to set a `pt` variable as we now longer have
+a mouse position. For now, we've set that to 100,100 like so which is just a
+fixed point for the dots to rotate around.
+
+Once more build the project in mill or sbt and then run `yarn start`. This time
+you'll notice that clicking the game doesn't do anything, but clicking the
+button adds a new rotating dot.
```

Comment:
Minor thing: If you just `yarn start` somewhere, then doing the scala build in a different terminal tab will do a hot-reload?

Reply:
Yup, but I think it's easier to follow as a tutorial if you're following steps, rather than switching tabs. Might be just personal preference... I just find it easier